### ISSUE TITLE
1.3.1 Info and Relationships Vs. 1.3.2 Meaningful Sequence

### ISSUE THREAD

**Issue by jake-abma** [w3c/wcag#1519]
I'm wondering what 1.3.2 has that is not covered by 1.3.1

1.3.1 clearly covers "structure, and relationships conveyed through presentation" and "programmatically determined"

1.3.2 is about a "correct reading sequence" BUT isn't that covered already by "structure, and relationships conveyed through presentation" of 1.3.1?

**Comment by patrickhlauke** [comment]
are there situations where a certain visual presentation does not necessarily imply a particular reading order? thinking various floating boxes/sections, that reorder at different viewport sizes etc, which in isolation are properly structured, but where there's no implied order...and 1.3.2 would catch this situation?

**Comment by JAWS-test** [comment]
1.3.1 is formulated so general, that in my opinion also the reading order belongs to it. But since there is a separate SC for reading order, I include all problems there. If there was no 1.3.2, I would consider problems with reading sequence at 1.3.1. Analog SC 2.1.1 and 2.1.2 - if there was no separate SC for keyboard trap, I would always consider it a problem at 2.1.1.

**Comment by erikkroes** [comment]
Everything that is covered by 1.3.1 needs to be "programmatically determined or are available in text". 1.3.2 only mentions "can be programmatically determined".

So while the situation might be covered by 1.3.1, the solution needed to satisfy 1.3.2 can be more strict than 1.3.1.

**Comment by alastc** [comment]
The way I've always understood it is that 1.3.1 allows for a programmatically determined but wrong order, 1.3.2 does not.

For example, using CSS layout to do 3 'cards' which each have a correct heading, image, and description.
However, the order is heading/heading/heading, then image/image/image, then desc/desc/desc. Similar to [F49](https://w3c.github.io/wcag/techniques/failures/F49).

I guess you could read that aspect into 1.3.1, but I've always kept that as "does this element have the right markup" rather than ordering aspects.

**Comment by jake-abma** [comment]
@alastc I did too, BUT when re-reading the SC it clearly states: "structure, **and relationships** conveyed through presentation"

The AND RELATIONSHIPS covers it already (doesn't it?)

**Comment by alastc** [comment]
Yes, the relationship can be programmatically determined, but doesn't mean it is a _meaningful_ order.

Taking to a more extreme case, where someone has used the tab key in word to create columns, it is in the same structure but the order of the text loses meaning.

**Comment by jake-abma** [comment]
hmm, depends on the definition of 'relationship', if the order is not meaningful the relationship between the different items / text in the content is not clear as I see it. So if you have a name/value pair, they have a relationship and must be determinable, also for 1.3.1

the line is a bit blurry although 1.3.2 is more strict clear (but not sure if it fits the conditions for a SC as they should not be covered in another SC)

**Comment by jake-abma** [comment]
still wondering if someone can provide an example of an implementation where there in no meaningful sequence while there is  a clear **relationship** that can be programmatically determined.

> relationships
> meaningful associations between distinct pieces of content

**Comment by alastc** [comment]
I'll try this example:
![image](https://user-images.githubusercontent.com/216630/101343188-89440000-387b-11eb-9f4f-c33b49d1abf0.png)

NB: This is a hypothetical based on a screenshot, the site I used doesn't do it that way! 

So you have things that should be in 3 columns to make sense, but the layout is created in a way that renders them as rows so things line up. (Not really needed these days, but I occasionally come across it.) AT would read the three alt-texts (if not null), then titles, then costs, then colours, then rating & promo.

For assistive technology it is a problem. The order is programmatically available but meaningless.

Where 1.3.1 is looking for relationships like th/td and Hx, this one is just looking at the sequence of content.

**Comment by jake-abma** [comment]
Get that example @alastc , well aware, but I was just challenging the "relationships" part. The relationship is gone in this case BUT it also depends on the definition of what we mean with 'relationship'

> relationships
> meaningful associations between distinct pieces of content

Is the example you provide a "meaningful associations between distinct pieces of content"?

Visually 'yes' programmatically 'no', as in, can't be programmatically determined, so no need for 1.3.2 (?!)

But I'll close the issue, Silver will solve all our challenges... ;-)
